### PR TITLE
Agent add unique number

### LINKED ISSUE #273: build nvidia image error when use github action
**Describe the bug**
there has an error when build the image:
`Reading package lists...
W: GPG error: https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64  InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY A4B469963BF863CC
E: The repository 'https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64  InRelease' is not signed.
The command '/bin/sh -c curl -fsSL [https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/${NVARCH}/7fa2af80.pub](https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/$%7BNVARCH%7D/7fa2af80.pub) | apt-key add -     && echo "deb [https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/${NVARCH}](https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/$%7BNVARCH%7D) /" > /etc/apt/sources.list.d/cuda.list     && if [ ! -z ${NV_ML_REPO_ENABLED} ]; then echo "deb ${NV_ML_REPO_URL} /" > /etc/apt/sources.list.d/nvidia-ml.list; fi     && apt-get update     && apt-get install -y --no-install-recommends cuda-cudart-11-4=${NV_CUDA_CUDART_VERSION} ${NV_CUDA_COMPAT_PACKAGE}     && ln -s cuda-11.4 /usr/local/cuda     && apt-get clean all     && rm -rf /var/lib/apt/lists/* /tmp/*' returned a non-zero code: 100`

the reason is:
1. https://github.com/NVIDIA/nvidia-docker/issues/1631
2. https://forums.developer.nvidia.com/t/notice-cuda-linux-repository-key-rotation/212772

**To Reproduce**

1. add a tag, like v0.1.1
2. then look the step of build agent

**Expected behavior**
success

**Environment**
github action

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Description
fix #273 
## Modules
- [ ] UI
- [ ] Controller
- [x] Agent
- [ ] Client
- [ ] Python-SDK
- [x] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [ ] add necessary doc

### CODE REVIEW

**Review by anda-ren — DISMISSED** [review]
lgtm

File: docker/Dockerfile.agent [hunk 1]
```diff
@@ -30,9 +30,20 @@ ENV CUDA_VERSION 11.4.0
...
+#    && echo "deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/${NVARCH} /" > /etc/apt/sources.list.d/cuda.list \
+#    && if [ ! -z ${NV_ML_REPO_ENABLED} ]; then echo "deb ${NV_ML_REPO_URL} /" > /etc/apt/sources.list.d/nvidia-ml.list; fi \
+#    && apt-get update \
+#    && apt-get install -y --no-install-recommends cuda-cudart-11-4=${NV_CUDA_CUDART_VERSION} ${NV_CUDA_COMPAT_PACKAGE} \
+#    && ln -s cuda-11.4 /usr/local/cuda \
+#    && apt-get clean all \
+#    && rm -rf /var/lib/apt/lists/* /tmp/*
+
+## wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/cuda-keyring_1.0-1_all.deb |
+COPY external/cuda-keyring_1.0-1_all.deb cuda-keyring_1.0-1_all.deb
```
Comment (tianweidut) [inline docker/Dockerfile.agent:43]: failed to download cuda-keyring_1.0-1_all.deb ?

Comment (goldenxinxing) [reply docker/Dockerfile.agent:43]: yes

Comment (goldenxinxing) [reply docker/Dockerfile.agent:42]: try again and it works again，already update

File: docker/Dockerfile.agent [hunk 2]
```diff
@@ -30,9 +30,20 @@ ENV CUDA_VERSION 11.4.0
 ENV NV_ML_REPO_ENABLED 1
 ENV NV_ML_REPO_URL https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu2004/${NVARCH}
 
-RUN curl -fsSL https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/${NVARCH}/7fa2af80.pub | apt-key add - \
-    && echo "deb https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/${NVARCH} /" > /etc/apt/sources.list.d/cuda.list \
-    && if [ ! -z ${NV_ML_REPO_ENABLED} ]; then echo "deb ${NV_ML_REPO_URL} /" > /etc/apt/sources.list.d/nvidia-ml.list; fi \
+#RUN curl -fsSL https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/${NVARCH}/7fa2af80.pub | apt-key add - \
```
Comment (tianweidut) [inline docker/Dockerfile.agent:33]: remove 33-40 comment lines?

File: docker/Dockerfile.agent [hunk 3]
```diff
@@ -30,9 +30,20 @@ ENV CUDA_VERSION 11.4.0
...
+#    && apt-get install -y --no-install-recommends cuda-cudart-11-4=${NV_CUDA_CUDART_VERSION} ${NV_CUDA_COMPAT_PACKAGE} \
+#    && ln -s cuda-11.4 /usr/local/cuda \
+#    && apt-get clean all \
+#    && rm -rf /var/lib/apt/lists/* /tmp/*
+
+COPY external/cuda-keyring_1.0-1_all.deb cuda-keyring_1.0-1_all.deb
+RUN apt install -y --no-install-recommends dirmngr \
+    && wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/cuda-keyring_1.0-1_all.deb \
+    && dpkg -i cuda-keyring_1.0-1_all.deb \
+    && apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2004/x86_64/3bf863cc.pub \
```
[the remark this answers is not in the document]
Comment (goldenxinxing) [reply docker/Dockerfile.agent:36]: done

**Review by tianweidut — APPROVED** [review]
LGTM

**Pull request merged**